### PR TITLE
Sets a11y traversal order in android accessibility bridge

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/flutter/issues/83121

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: shell/platform/android/io/flutter/view/AccessibilityBridge.java [hunk 1]
```diff
@@ -2455,7 +2467,10 @@ private void updateRecursively(
         }
       }
 
+      int previousNodeId = -1;
       for (SemanticsNode child : childrenInTraversalOrder) {
+        child.previousNodeId = previousNodeId;
```
Comment (goderbauer) [inline shell/platform/android/io/flutter/view/AccessibilityBridge.java:2472]: It's guaranteed that we will run through this code again when the order changes and with that a child's previousNodeId changes?

**Review by goderbauer — COMMENTED** [review]
/cc @yjbanov It has been a while, but do you remember why we stopped using the setTraversalAfter API in https://github.com/flutter/engine/pull/5066?

File: shell/platform/android/io/flutter/view/AccessibilityBridge.java (hunk 1 above)
Comment (chunhtai) [reply shell/platform/android/io/flutter/view/AccessibilityBridge.java:2472]: yes, this method will called whenever the bridge receive update from framework.

File: shell/platform/android/io/flutter/view/AccessibilityBridge.java [hunk 2]
```diff
@@ -2455,7 +2467,10 @@ private void updateRecursively(
         }
       }
 
+      int previousNodeId = -1;
```
Comment (yjbanov) [inline shell/platform/android/io/flutter/view/AccessibilityBridge.java:2470]: Would be good to leave a comment explaining why traversal order of the child list alone is not sufficient on Android.

**Review by yjbanov — APPROVED** [review]
lgtm

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**